src/components/ManaCalculator/components/ManaCalculatorInput.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { JSX, useEffect, useState } from 'react';
+
+interface IManaCalculatorCustomInput {
+  value: string | number;
+  onChange: (value: string) => void;
+  className?: string;
+  min?: number;
+  max?: number;
+}
+
+export function ManaCalculatorInput({
+  value,
+  onChange,
+  className,
+  min,
+  max,
+}: IManaCalculatorCustomInput): JSX.Element {
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string | number>(value);
```

Comment:
Why do you need another useState for the input value? You are already getting the value via the `value` prop

Reply:
I did this so the input shows a value, either is invalid or not, and then, after verifying that the function doesn't change, the state of the passed value gets updated. If we don't do this the user wont be able to see incorrectly typed characters ( a `2` in a `min 0` `max 1` input )